strconv/uint_test.mbt

```diff
@@ -0,0 +1,92 @@
+  // Different bases with valid input
+  inspect!(@strconv.parse_uint64?("FF", base=16), content="Ok(255)")
+  inspect!(@strconv.parse_uint64?("0xFF"), content="Ok(255)")
+  inspect!(@strconv.parse_uint64?("377", base=8), content="Ok(255)")
+  inspect!(@strconv.parse_uint64?("0o377"), content="Ok(255)")
+  inspect!(@strconv.parse_uint64?("11111111", base=2), content="Ok(255)")
+  inspect!(@strconv.parse_uint64?("0b11111111"), content="Ok(255)")
+}
+
+test "@strconv.parse_uint64/underscore" {
+  // Valid underscore placements
+  inspect!(@strconv.parse_uint64?("1_000_000"), content="Ok(1000000)")
+  inspect!(@strconv.parse_uint64?("0xff_ff_ff"), content="Ok(16777215)")
+  // Invalid underscore placements
+  inspect!(@strconv.parse_uint64?("_123"), content="Err(invalid syntax)")
+  inspect!(@strconv.parse_uint64?("123_"), content="Err(invalid syntax)")
+  inspect!(@strconv.parse_uint64?("1__23"), content="Err(invalid syntax)")
+}
+
+test "panic @strconv.parse_uint64/errors" {
+  // Empty string
+  ignore(@strconv.parse_uint64!(""))
+  // Invalid base
+  ignore(@strconv.parse_uint64!("123", base=37))
+  // Signs not allowed
+  ignore(@strconv.parse_uint64!("+123"))
+  ignore(@strconv.parse_uint64!("-123"))
+  // Range error (overflow)
+  ignore(@strconv.parse_uint64!("18446744073709551616"))
+}
+
+test "@strconv.parse_uint64/hex_and_edge_cases" {
+  // Valid hexadecimal numbers with 0x/0X prefix
+  inspect!(@strconv.parse_uint64?("0xDEADBEEF"), content="Ok(3735928559)")
+  inspect!(@strconv.parse_uint64?("0XDEADBEEF"), content="Ok(3735928559)")
+
+  // Valid hexadecimal numbers without prefix when base=16 is specified
+  inspect!(
+    @strconv.parse_uint64?("deadbeef", base=16),
+    content="Ok(3735928559)",
+  )
+
+  // Valid hexadecimal numbers with allowed underscore placements
+  inspect!(@strconv.parse_uint64?("0xDEAD_BEEF"), content="Ok(3735928559)")
+  inspect!(
+    @strconv.parse_uint64?("dead_beef", base=16),
+    content="Ok(3735928559)",
+  )
+  inspect!(@strconv.parse_uint64?("0x1_2345"), content="Ok(74565)")
+
+  // Edge case: zero value
+  inspect!(@strconv.parse_uint64?("0x0"), content="Ok(0)")
+  inspect!(@strconv.parse_uint64?("0x00"), content="Ok(0)")
+
+  // Edge case: maximum valid uint64 value (all bits set)
+  inspect!(
+    @strconv.parse_uint64?("0xFFFFFFFFFFFFFFFF"),
+    content="Ok(18446744073709551615)",
+  )
+  inspect!(@strconv.parse_uint64?("0xDEADBEEF_"), content="Err(invalid syntax)")
+  inspect!(
+    @strconv.parse_uint64?("0xDE__AD_BEEF"),
+    content="Err(invalid syntax)",
+  )
+
+  // Invalid hex digit (G is not a valid base-16 digit)
+  inspect!(@strconv.parse_uint64?("0xDEADBEG"), content="Err(invalid syntax)")
+}
+
+test "edge cases" {
+  // Invalid: Missing digits after hex prefix
+  // inspect!(@strconv.parse_uint64?("0x"), content="Err(invalid syntax)")
+
+  // Invalid underscore placements in hexadecimal strings
+  // inspect!(@strconv.parse_uint64?("0x_DEADBEEF"), content="Err(invalid syntax)")
+
```

Comment:
what's the desired behavior?

Reply:
I'm not aware of a standard way for parsing the hex literals. So I think we could just follow other languages. This is valid in both Python and Go. See https://go.dev/play/p/jWnuL-QCwF- and 
<img width="621" alt="image" src="https://github.com/user-attachments/assets/7568fbeb-98cf-426e-a268-b4dd97b9840d" />

I will add more documentation though.

Reply:
note in both python and go:
```
>>> int("0x_DAE_",0)
Traceback (most recent call last):
  File "<python-input-3>", line 1, in <module>
    int("0x_DAE_",0)
    ~~~^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 0: '0x_DAE_'
```

Reply:
Yes, trailing underscore is handled. See https://github.com/moonbitlang/core/pull/1586/files#diff-f33bf2098e6c91bdc93d15281b2f928dfe14c05b8b5158fc4caf1ab98404e909R249